src/components/CounterInput/index.js

```diff
@@ -0,0 +1,243 @@
+import React, { useRef } from 'react';
+import PropTypes from 'prop-types';
+import RenderIf from '../RenderIf';
+import RelativeElement from '../Structural/relativeElement';
+import Label from '../Input/label';
+import HelpText from '../Input/styled/helpText';
+import ErrorText from '../Input/styled/errorText';
+import { uniqueId } from '../../libs/utils';
+import PlusIcon from './icons/plus';
+import MinusIcon from './icons/minus';
+import floatify from './helpers/floatify';
+import isValueNan from './helpers/isValueNan';
+import disableButton from './helpers/disableButtons';
+import { StyledContainer, StyledInput, StyledButton, ButtonContainer } from './styled';
+
+/**
+ * CounterInput allows to increase or decrease a numerical value.
+ * @category Form
+ */
+
+export default function CounterInput(props) {
+    const {
+        id,
+        name,
+        value,
+        placeholder,
+        onChange,
+        tabIndex,
+        onFocus,
+        onBlur,
+        onClick,
+        onKeyDown,
+        disabled,
+        readOnly,
+        required,
+        min,
+        max,
+        isBare,
+        error,
+        variant,
+        step,
+        label,
+        labelAlignment,
+        hideLabel,
+        bottomHelpText,
+        className,
+        style,
+    } = props;
```

Comment:
since this is an inputable component and we need to integrate it with redux-form, props must be passed through the useReduxForm hook, take a look at other components like the FileSelector